scheduler/scheduler.go

```diff
@@ -755,13 +812,14 @@ func (driver *MesosSchedulerDriver) start() (mesos.Status, error) {
 		})
 	})
 
-	// register with Detect() AFTER we have a self pid from the messenger, otherwise things get ugly
-	// because our internal messaging depends on it. detector callbacks are routed over the messenger
-	// bus, maintaining serial (concurrency-safe) callback execution.
-	log.V(1).Infof("starting master detector %T: %+v", driver.masterDetector, driver.masterDetector)
-	driver.masterDetector.Detect(listener)
-
-	log.V(2).Infoln("master detector started")
+	if driver.masterDetector != nil {
```

Comment:
making masterDetector optional keeps setup for unit tests simpler, but i'm not sure that I like the checks for nil in this class. i'd almost rather have a NoopDetector implementation that unit tests used (guaranteed to do nothing all the time).